File: backend/internal/model/campaign.go
package model

import "time"

type Campaign struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	OwnerId     string    `json:"owner_id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CampaignMember struct {
	CampaignID string    `json:"Campaign_id"`
	UserID     string    `json:"user_id"`
	Role       string    `json:"role"` // "owner", "editor", "viewer"
	JoinetAt   time.Time `json:"joined_at"`
}

File: frontend/src/lib/types/Campaign.ts
import { Members } from "./Members";

export interface Campaign {
  id: string;
  name: string;
  description?: string;
  ownerID: string;
  created_at: Date;
  updated_at: Date;
}

export interface CampaignWithRole extends Campaign {
  role: string;
}

export interface CampaignDetailsResponse extends Campaign {
  members: Members[];
}
